staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/structuredmerge.go

```diff
@@ -60,13 +61,14 @@ func NewStructuredMergeManager(models openapiproto.Models, objectConverter runti
 		updater: merge.Updater{
 			Converter: internal.NewVersionConverter(typeConverter, objectConverter, hub),
 		},
+		resetFields: resetFields,
 	}, nil
 }
 
 // NewCRDStructuredMergeManager creates a new Manager specifically for
 // CRDs. This allows for the possibility of fields which are not defined
 // in models, as well as having no models defined at all.
-func NewCRDStructuredMergeManager(models openapiproto.Models, objectConverter runtime.ObjectConvertor, objectDefaulter runtime.ObjectDefaulter, gv schema.GroupVersion, hub schema.GroupVersion, preserveUnknownFields bool) (_ Manager, err error) {
+func NewCRDStructuredMergeManager(models openapiproto.Models, objectConverter runtime.ObjectConvertor, objectDefaulter runtime.ObjectDefaulter, gv schema.GroupVersion, hub schema.GroupVersion, preserveUnknownFields bool, resetFields *fieldpath.Set) (_ Manager, err error) {
```

Comment:
Can we implement this as a layer around the structuredManager? I'd like to keep the number of parameters here limited.

Reply:
OK we can't because we need to do that on typed objects.